uefi/uefi_test.go

```diff
@@ -30,3 +31,136 @@ func TestUnmarshalTypedFirmware(t *testing.T) {
 		t.Errorf("got %q, expected CHARLIE", outSection.Name)
 	}
 }
+
+var (
+	// Checksum Tests
+	emptyBuf  = []byte{}
+	sampleBuf = []byte{1, 2, 3, 4}
+	overBuf   = []byte{0x1, 0x2, 0xFF, 0xFF}
+	zeroBuf   = []byte{0, 0, 0, 0}
+	threeBuf  = []byte{3, 3, 3}
+)
+
+func TestChecksum8(t *testing.T) {
+	var tests = []struct {
+		name string
+		buf  []byte
+		res  uint8
+	}{
+		{"emptyBuf", emptyBuf, 0},
+		{"sampleBuf", sampleBuf, 10},
+		{"overBuf", overBuf, 0x1},
+		{"zeroBuf", zeroBuf, 0},
+		{"threeBuf", threeBuf, 9},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			if res := Checksum8(test.buf); res != test.res {
+				t.Errorf("Checksum8 wrong result!, input was %#x, wanted \n%#x\n, got \n%#x\n", test.buf, test.res, res)
+			}
+		})
+	}
+}
+
+func TestChecksum16(t *testing.T) {
+	var tests = []struct {
+		name string
+		buf  []byte
+		res  uint16
+		msg  string
+	}{
+		{"emptyBuf", emptyBuf, 0, ""},
+		{"sampleBuf", sampleBuf, 0x46, ""},
+		{"overBuf", overBuf, 0x11, ""},
+		{"zeroBuf", zeroBuf, 0, ""},
+		{"threeBuf", threeBuf, 0, fmt.Sprintf("byte slice does not have even length, not able to do 16 bit checksum. Length was %v",
+			len(threeBuf))},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			res, err := Checksum16(test.buf)
+			if res != test.res {
+				t.Errorf("Checksum16 wrong result!, input was %#x, wanted \n%#x\n, got \n%#x\n", test.buf, test.res, res)
+			}
+			if err == nil && test.msg != "" {
+				t.Errorf("Error was not returned, expected %v", test.msg)
+			} else if err != nil && err.Error() != test.msg {
+				t.Errorf("Mismatched Error returned, expected \n%v\n got \n%v\n", test.msg, err.Error())
+			}
+		})
+	}
+}
+
+func TestWrite3Size(t *testing.T) {
+	var tests = []struct {
+		name string
+		val  uint64
+		res  [3]byte
+	}{
+		{"emptySize", 0x0, [3]byte{0, 0, 0}},
+		{"sampleSize", 0xABCDEF, [3]byte{0xEF, 0xCD, 0xAB}},
+		{"max3ByteSize", 0xFFFFFF, [3]byte{0xFF, 0xFF, 0xFF}},
+		{"over3ByteSize", 0x1000000, [3]byte{0xFF, 0xFF, 0xFF}},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
Remove `t.Parallel()`. It is creating false positives.

Reply:
Done